src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -20,27 +24,50 @@ public abstract class OwnedBuffer<T> : IDisposable, IKnown
 
         public virtual BufferHandle Pin(int index = 0)
```

Comment:
Should we make the method unsafe? Will a class that overrides this method ever be safe?